content/posts/unpublished/styling-your-legacy-micro-frontends/post.md

```diff
@@ -0,0 +1,135 @@
+
+Including coupled packages results in a significant amount of code duplication, but once it's set and configured you can nearly forget about making further changes to your CSS.
+
+### Set-up
+
+1. Create a `/themes` folder at the root level.
+2. Go to the newly created folder and install UI-Ang with its dependencies.
+```npm
+npm i @backbase/ui-ang@10 ngx-quill@16
+```
+3. Copy `/backbase-theme-preset-template` from the `/node_modules/@backbase/ui-ang/dist` to the `/themes` folder.
+4. Update the `/shell/src/styles.scss` to include the new theme.
+```scss
+@import '@backbase/backbase-theme-business-preset/scss/main';
+
+.angular17-theme {
+  @import '../../themes/backbase-theme-preset-template/scss/main';
+}
+```
+5. Don't forget to update your remote `AppComponent`.
+```angular2html
+<div class="angular17-theme">
+  <router-outlet></router-outlet>
+</div>
+```
+
+In an ideal world this should be enough to have both themes working side by side, but the compiler throws some errors because of the incompatibilities of using two Sass versions.
+
+![](assets/sass_error_bootstrap_buttons.jpg)
+
+`"sass"` and `"sass-loader"` are dependencies of `"@angular-devkit/build-angular"` so there is not much you can do as the legacy app must stay in Angular 12. Then it's time for you to update manually the source scss files.
+
+1. Copy the `/scss` folder from `/themes/node_modules/bootstrap` to a new `/themes/bootstrap5` directory.
+2. Do the same for UI-Ang copying the `/scss` and `/assets` folder from `/themes/node_modules/@backbase/ui-ang` to a new `/themes/ui-ang-10` directory.
+3. Search for all the references to bootstrap imports inside `/themes/ui-ang-10` and replace them to point to the custom `/themes/bootstrap5` one. For example, `@import "bootstrap/scss/mixins";` is now `@import "../../../bootstrap5/scss/mixins";`. Make sure to adjust all the relative paths.
+4. Now do the same inside `/themes/backbase-theme-preset-template` and replace the references to UI-Ang. For example, `@import "~@backbase/ui-ang/scss/1-variables/variables";` with `@import "../../../ui-ang-10/scss/1-variables/variables";`.
+5. Check that the compiler still throws the same error as before starting with the manual update, or fix the imports otherwise.
+6. In the `/themes/bootstrap5/scss/mixins/_buttons.scss` the offending line is the 27 `--#{$prefix}btn-focus-shadow-rgb: #{to-rgb(mix($color, $border, 15%))};` so you need to comment it out. This makes the build pass but throws a new error.
+   ![](assets/sass_error_accounts_list.jpg)
+7. In the `/themes/ui-ang-10/scss/5-components/journeys/accounts-journey/_accounts-accounts-list.scss` the line failing it's the 13, so you can replace it with `$accounts-list-cards-column-min-width: max(18.5rem, 25%);`.
```

Comment:
Can we not refer to specific lines? Readers can be either people doing a step-by-step or people reading for their knowledge, and the latter will likely get confused with that – also, if a different patch version of the libraries might differ here. From the blog perspective, I'd rather keep it more generic